### PR TITLE
#5983 Cleanup black box FMUs

### PR DESCRIPTION
**Pull request by arun3688**
### Purpose
This cleans up the files in FMU resource directory,  when used with --fmiFilter=blackBox, see https://trac.openmodelica.org/OpenModelica/ticket/5983

### changed Files, 
SerializeModelInfo.mo  -- generate the _info.json in resource directory, if fmiFilter=blackBox not set
CevalScriptBackend.mo -- generate the static.log in resource directory, if fmiFilter=blackBox not set
CodeGenFmu.tpl   -- generate .summary file in resource directory, if fmiFilter=blackBox not set

### CODE REVIEW

File: OMCompiler/Compiler/Template/CodegenFMU.tpl [hunk 1]
```diff
@@ -1166,7 +1166,9 @@ match platform
   <%\t%>$(LD) -o <%modelNamePrefix%>$(DLLEXT) $(OFILES) $(RUNTIMEFILES) <%dirExtra%> <%libsPos1%> @BSTATIC@ <%libsPos2%> @BDYNAMIC@ $(LDFLAGS)
   <%\t%>cp <%fileNamePrefix%>$(DLLEXT) <%fileNamePrefix%>_FMU.libs ../binaries/$(FMIPLATFORM)/
   endif
+  if not Flags.getConfigEnum(Flags.FMI_FILTER) == Flags.FMI_BLACKBOX then
```
Comment (adrpo) [inline OMCompiler/Compiler/Template/CodegenFMU.tpl:1169]: This is not correct!
it should be 
<%if not Flags.getConfigEnum(Flags.FMI_FILTER) == Flags.FMI_BLACKBOX then
'<%\t%>head -n20 Makefile > ../resources/$(FMIPLATFORM).summary'
%>

Comment (arun3688) [reply OMCompiler/Compiler/Template/CodegenFMU.tpl]: Thanks adrian, this is fixed with this https://github.com/OpenModelica/OpenModelica/pull/906/commits/1d88860fcd180400750ed5fc38dd6966d25b444a

File: OMCompiler/Compiler/Script/CevalScriptBackend.mo [hunk 2]
```diff
@@ -3805,17 +3805,17 @@ algorithm
     if isWindows then
       configureLogFile := "\""+configureLogFile+"\"";
     end if;
-    // do not create static.log in resource directory when --fmiFilter=blackBox
-    if intLt(Flags.getConfigEnum(Flags.FMI_FILTER), Flags.FMI_BLACKBOX) then
-      configureFMU(platform, fmutmp, configureLogFile, isWindows);
+    configureFMU(platform, fmutmp, configureLogFile, isWindows);
+    if Flags.getConfigEnum(Flags.FMI_FILTER) == Flags.FMI_BLACKBOX then
+      System.removeFile(configureLogFile);
```
Comment (arun3688) [inline OMCompiler/Compiler/Script/CevalScriptBackend.mo:3810]: system.removeFile(configureLogFile) , does not remove the static.log file in windows,  why are we running the configureFMU() if the blackBox option is set,  it is just creating this static.log right ?

Comment (arun3688) [reply OMCompiler/Compiler/Script/CevalScriptBackend.mo:3810]:
```
 cmd := "rm " + configureLogFile;
 if 0 <> System.systemCall(cmd, outFile=logfile) then
    Error.addInternalError("Failed to remove Json file: " + fmutmp , sourceInfo());
 end if ;
```
However using the cmd "rm  " works

Comment (arun3688) [reply OMCompiler/Compiler/Script/CevalScriptBackend.mo:3810]: @adrpo @sjoelund  why this "System.removeFile()" not working in mingw ?

Comment (adrpo) [reply OMCompiler/Compiler/Script/CevalScriptBackend.mo:3807]: Damn, it includes the quotes, that's why it cannot find the file.

**Pull request merged**